### ISSUE TITLE
KOReader crashes when downloading from OPDS server (catalog)

### ISSUE THREAD

**Issue by IDontKnow218** [koreader/koreader#11216]
* KOReader version: v2023.10-51-g13a521c39_2023-12-08
* Device: Kindle HD 8 (2022) (KFRAWI)

#### Issue

When I try to download epubs off my OPDS server, it gives me a message that it would take some time to complete. A few seconds later, KOReader crashes.

#### Steps to reproduce

1. Navigate to OPDS server.
2. Download anything.
3. Get message
4. Crash occurs
 
##### `crash.log` (if applicable)
`crash.log` is a file that is automatically created when KOReader crashes. It can normally be found in the KOReader directory:

Here is the [crash.log](https://github.com/koreader/koreader/files/13628813/crash.log) file

* `/mnt/private/koreader` for Cervantes
* `koreader/` directory for Kindle
* `.adds/koreader/` directory for Kobo
* `applications/koreader/` directory for Pocketbook

Android logs are kept in memory. Please go to [Menu] → Help → Bug Report to save these logs to a file.

Please try to include the relevant sections in your issue description.
You can upload the whole `crash.log` file (zipped if necessary) on GitHub by dragging and dropping it onto this textbox.

If your issue doesn't directly concern a Lua crash, we'll quite likely need you to reproduce the issue with *verbose* debug logging enabled before providing the logs to us.
To do so, from the file manager, go to [Tools] → More tools → Developer options, and tick both `Enable debug logging` and `Enable verbose debug logging`.
You'll need to restart KOReader after toggling these on.

If you instead opt to inline it, please do so behind a spoiler tag:
<details>
  <summary>crash.log</summary>

```
<Paste crash.log content here>
```
</details>

**Comment by Frenzie** [comment]
What exactly do you mean by "crashes"? On Android you might currently get a "not responding" while it's busy.

**Comment by IDontKnow218** [comment]
> What exactly do you mean by "crashes"? On Android you might currently get a "not responding" while it's busy.

You're right. It doesn't respond then crashes, but sometimes it crashes immediately.

**Comment by Frenzie** [comment]
Could you activate verbose debug logging as described in the issue template?

**Comment by IDontKnow218** [comment]
Here you go [crash.log](https://github.com/koreader/koreader/files/13635981/crash.log)

**Comment by Frenzie** [comment]
If you're familiar with adb maybe something can be obtained that way. Pinging @pazos
